File: android/src/main/java/com/margelo/nitro/nitrovideo/records/VideoThumbnailOptions.kt
package com.margelo.nitro.nitrovideo.records

import java.io.Serializable

class VideoThumbnailOptions(
    val maxWidth: Int? = null,
    val maxHeight: Int? = null
) : Serializable {
    fun toNativeSizeLimit(): Pair<Int, Int>? {
        if (this.maxWidth == null && this.maxHeight == null) {
            return null
        }
        val width = this.maxWidth ?: Int.MAX_VALUE
        val height = this.maxHeight ?: Int.MAX_VALUE

        require(width >= 1 && height >= 1) {
            "Failed to generate a thumbnail: The maxWidth and maxHeight parameters must be greater than zero"
        }
        return width to height
    }
}

File: ios/Thumbnails/VideoThumbnailOptions.swift
import CoreGraphics
import Foundation

public struct VideoThumbnailOptions: Codable, Sendable {
  public static let `default` = VideoThumbnailOptions()

  public var maxWidth: Int = 0
  public var maxHeight: Int = 0

  public init(maxWidth: Int = 0, maxHeight: Int = 0) {
    self.maxWidth = maxWidth
    self.maxHeight = maxHeight
  }

  public func getMaxSize() -> CGSize {
    return CGSize(width: maxWidth, height: maxHeight)
  }
}
